subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/ResolutionErrorRenderer.java

```diff
@@ -110,13 +109,11 @@ private void registerError(Action<StyledTextOutput> errorAction) {
         errorActions.add(errorAction);
     }
 
-    private boolean hasVersionConflictOnRequestedDependency(List<ModuleVersionIdentifier> moduleVersionIdentifiers) {
-        boolean matchesSpec = false;
-        for (final ModuleVersionIdentifier mvi : moduleVersionIdentifiers) {
-            matchesSpec |= dependencySpec.isSatisfiedBy(new DependencyResult() {
+    private boolean hasVersionConflictOnRequestedDependency(final ModuleIdentifier moduleIdentifier) {
+        return dependencySpec.isSatisfiedBy(new DependencyResult() {
                 @Override
                 public ComponentSelector getRequested() {
-                    return DefaultModuleComponentSelector.newSelector(mvi.getModule(), mvi.getVersion());
+                    return DefaultModuleComponentSelector.newSelector(moduleIdentifier, "unknown");
```

Comment:
:o: I'm not sure I understand this change, can you explain?

Reply:
Versions are not expected in the dependency declaration of `dependencyInsight` and so instead of doing multiple matches, I do a singe one on the module id with an "unknown" version.
Ideally I would be able to express this, but the types involved do not make that obvious.

Reply:
Damn thinking some more about it,  I realize that saying `./gradlew dependencyInsight --configuration conf --dependency org.test:foo:1.4` is not invalid and would have matched if `org.test:foo:1.4` was a version under conflict but would not longer match anymore.
Any suggestion?

Reply:
The motivation behind that change is that `FailOnVersionConflictArtifactsVisitor` assumed that all conflicts were about a list of versions. But with capabilities coming into play that's no longer the case.
One idea would be to craft a list of versions for a capability conflict, for the sole purpose of matching requested dependencies.